src/io/xpi.js

```diff
@@ -140,7 +142,8 @@ export class Xpi extends IOBase {
 
   getChunkAsBuffer(path, chunkLength) {
     return new Promise((resolve, reject) => {
-      this.checkPath(path);
+      const pathErr = this._checkPath(path);
+      if (pathErr) return reject(pathErr);
```

Comment:
I think from a semantic point of view I'd still like to have `checkPath` raise an exception but use `try/except` here to catch the exception and then `reject` it properly. I also don't think we should rename `checkPath` to `_checkPath` but that's nothing to worry about too much.